src/System.Text.Json/src/System/Text/Json/JsonDocument.CustomDb.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json
+{
+    partial class JsonDocument
+    {
+        private struct CustomDb : IDisposable
```

Comment:
It's not immediately clear what format is being used for the data this stores... a good comment would be helpful.